### ISSUE TITLE
[noderesourcetopology] incorrect accounting of hostlevel resources (ephemeral storage) with scope=container

### ISSUE THREAD

**Issue by ffromani** [kubernetes-sigs/scheduler-plugins#751]
### Area

- [X] Scheduler
- [ ] Controller
- [ ] Helm Chart
- [ ] Documents

### What happened?

After #735 and #646 we performed further investigation in how we handle host-level resources, like ephemeral storage.
"host-level resources" are resources which don't have NUMA affinity so don't contribute to the noderesourcetopology decisions, at least in theory.
In #646 we added a fix to ignore the ephemeral storage (being a common example of hostlevel resource) in the filtering proper, which was indeed a needed fix; this however uncovered another issue in how we do resource accounting for hostlevel resources; When dealing with multiple containers, each requiring ephemeral storage, a couple of flaws in the `subtractFromNUMA` function causes:
1. When processing the first container the resource to be incorrectly subtracted from the counters for a NUMA cell. But since it is a hostlevel resource, the counters go negative, an [occurrence the code warns for, but does not treat as error](https://github.com/kubernetes-sigs/scheduler-plugins/blob/v0.28.9/pkg/noderesourcetopology/filter.go#L243)
2. When processing the subsequent containers, we now have unexpected entries in the per-NUMA resource counters, which cause [the guard to fail](https://github.com/kubernetes-sigs/scheduler-plugins/blob/v0.28.9/pkg/noderesourcetopology/filter.go#L120) and, since the resource cannot have NUMA affinity, to end up with [incorrect NUMA availability bitmask, leading to bad scheduling decision](https://github.com/kubernetes-sigs/scheduler-plugins/blob/v0.28.9/pkg/noderesourcetopology/filter.go#L142)

The proper fix is to review the way we treat these resources from the ground up, adding explicit handling all along the way.

### What did you expect to happen?

host-level resources like ephemeral storage should never contribute to the noderesourcetopology scheduling decisions

### How can we reproduce it (as minimally and precisely as possible)?

1. configure topology manager with scope=container
2. make sure the kubelet exposes the ephemeral storage resource
3. create a pod which have multiple (2+) container each of them requires ephemeral storage; set the overall resource requests in such a way that there is suitable worker nodes
4. observe the scheduler incorrectly discarding all the suitable worker nodes

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Scheduler Plugins version

<details>

</details>
